docs/docsite/rst/dev_guide/overview_architecture.rst

```diff
@@ -67,8 +76,65 @@ Here's what a simple playbook looks like::
     - common
     - content
 
+.. _ansible_search_path:
+
+The Ansible search path
+=======================
+
+Modules, module utilities, plugins, playbooks, and roles can live in multiple locations. If you
+write your own code to extend Ansible's core features, you may have multiple files with similar or the same names in different locations on your Ansible control node. The search path determines which of these files Ansible will discover and use on any given playbook run.
+
+Ansible's search path grows incrementally over a run. As
+Ansible finds each playbook and role included in a given run, it appends
+any directories related to that playbook or role to the search path. Those
+directories remain in scope for the duration of the run, even after the playbook or role
+has finished executing. Ansible loads modules, module utilities, and plugins in this order:
+
+1. Directories adjacent to a playbook specified on the command line. If you run Ansible with ``ansible-playbook /path/to/play.yml``, Ansible appends these directories if they exist:
+
+   .. code-block:: bash
+
+      /path/to/modules
+      /path/to/module_utils
+      /path/to/plugins
+
+2. Directories adjacent to a playbook that is statically imported by a
+   playbook specified on the command line. If ``play.yml`` includes
+   ``- import_playbook: /path/to/subdir/play1.yml``, Ansible appends these directories if they exist:
+
+   .. code-block:: bash
+
+      /path/to/subdir/modules
+      /path/to/subdir/module_utils
+      /path/to/subdir/plugins
+
+3. Subdirectories of a role directory referenced by a playbook. If
+   ``play.yml`` runs ``myrole``, Ansible appends these directories if they exist:
+
+   .. code-block:: bash
+
+      /path/to/roles/myrole/modules
+      /path/to/roles/myrole/module_utils
+      /path/to/roles/myrole/plugins
+
+4. Directories specified in ``ansible.cfg`` or by the various
+   ``DEFAULT_*_PATH`` environment variables, including the paths for the
+   various plugin types:
+
+   .. code-block:: bash
+
+      DEFAULT_MODULE_PATH
+      DEFAULT_MODULE_UTILS_PATH
+      DEFAULT_CACHE_PLUGIN_PATH
+      DEFAULT_FILTER_PLUGIN_PATH
```

Comment:
actually, ANSIBLE_LIBRARYis the env var for DEFAULT_MODULE_PATH, same for the others, you are listing the 'config names' not the environment variables.

Reply:
(which we should probably update internally to *not* use `DEFAULT_` prefix, since we've abolished that practice for new stuff and user code *shouldn't* probably be consuming those directly in most cases)

Reply:
I added references to both the env vars and the config names, with a link to the main config page for more info. If/when we update the code not to use the `DEFAULT_` prefix, we can update this page again . . . but meanwhile perfection is the enemy of improvement.